src/Licensing/ThroughputCollectorHostBuilderExtensions.cs

```diff
@@ -0,0 +1,24 @@
+﻿namespace Particular.License
+{
+    using Microsoft.Extensions.DependencyInjection;
+    using Microsoft.Extensions.Hosting;
+    using Particular.License.Contracts;
+    using Particular.License.Throughput.Audit;
+    using Particular.License.Throughput.Broker;
+
+    public static class ThroughputCollectorHostBuilderExtensions
+    {
+        public static IHostApplicationBuilder AddThroughputCollector(this IHostApplicationBuilder hostBuilder, string broker, string serviceControlAPI, string errorQueue, string auditQueue, string transportConnectionString, string persistenceType)
+        {
+            var services = hostBuilder.Services;
+            services.AddSingleton(new PlatformData { Broker = broker, ServiceControlAPI = serviceControlAPI, AuditQueue = auditQueue, ErrorQueue = errorQueue, TransportConnectionString = transportConnectionString, PersistenceType = persistenceType });
+            services.AddHostedService<AuditThroughputCollectorHostedService>();
+            services.AddHostedService<BrokerThroughputCollectorHostedService>();
+
```

Comment:
I'm not injecting the monitoring one here as yet, as we don't know which way that will work - ie the queue or the added persistence.